packages/block-editor/src/hooks/lock.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { has } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Filters registered block settings, extending attributes to include `lock`.
+ *
+ * @param {Object} settings Original block settings.
+ *
+ * @return {Object} Filtered block settings.
+ */
+export function addAttribute( settings ) {
+	// allow blocks to specify their own attribute definition with default values if needed.
+	if ( has( settings.attributes, [ 'lock', 'type' ] ) ) {
+		return settings;
+	}
+	// Gracefully handle if settings.attributes is undefined.
+	settings.attributes = {
+		...settings.attributes,
+		lock: {
+			type: 'object',
+		},
+	};
+
+	return settings;
+}
+
+addFilter( 'blocks.registerBlockType', 'core/lock/addAttribute', addAttribute );
```

Comment:
Why is this a hook if the logic is built-in into the selectors of the block-editor package, should we just move this logic to the registerBlockType function (I guess both server side and client side)

Reply:
I'm not sure if GB is departing away from using filters or not. In my testing, removing this filter doesn’t break anything, having it provides a sensible default. But if we're going to double down on this, it might be worth moving it to registration process.

Currently, we use this hook to [detect native usage and disable](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/js/blocks/cart-checkout/checkout-i2/hacks.ts#L49) our hacks file that does approximately this.

Reply:
Removing the filter means blocks won't persist their "lock" attribute upon modification (which is a breakage).

What I'm saying is not related about whether we're moving away from filters or not, I'm saying that since "locks" are developed as a "build-in" feature in selectors... the filter shouldn't be removable, so it shouldn't be a filter and should be added directly to the registration function IMO. Not a big deal though.

Reply:
This is a good point by Riad I guess in this case we may remove the filter and include the equivalent code without the filter given that removing the filter in fact causes a wrong behavior.

Reply:
I'm sorry for missing this as I was the one that suggested the filter initially.

Reply:
cc: @senadir